generator/golang/resolver.go

```diff
@@ -71,7 +71,7 @@ func (r *Resolver) ResolveFieldTypeName(f *parser.Field) (TypeName, error) {
 		return "", err
 	}
 
-	if NeedRedirect(f) {
+	if NeedRedirect(f) && !(r.util.Features().EnableRefInterface && isRefInterfaceType(r.root, f.Type)) {
```

Comment:
Don't Repeat Yourself. Please make it a function, which is also eaiser to understand.